src/Arriba/Arriba.Server/Extensions/OAuthExtension.cs

```diff
@@ -28,9 +32,28 @@ public static IServiceCollection AddOAuth(this IServiceCollection services, IArr
                 auth.DefaultPolicy = jwtBearerPolicy;
             });
 
-            services.AddSingleton((_) => serverConfig.OAuthConfig);
-
             return services;
         }
     }
+
+    public class AllowAnonymous : IAuthorizationHandler
+    {
+        private readonly IArribaServerConfiguration _serverConfiguration;
+
+        public AllowAnonymous(IArribaServerConfiguration serverConfiguration)
+        {
+            _serverConfiguration = serverConfiguration;
+        }
+
+        public Task HandleAsync(AuthorizationHandlerContext context)
+        {
+            if (!_serverConfiguration.EnabledAuthentication)
+            {
+                foreach (IAuthorizationRequirement requirement in context.PendingRequirements.ToList())
```

Comment:
Does this only work if it's the first or last handler?

Reply:
[Filters in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/mvc/controllers/filters?view=aspnetcore-3.1 )
Authorization filters run first and are used to determine whether the user is authorized for the request. Authorization filters short-circuit the pipeline if the request is not authorized.

To use the [Authorize] attribute on the Controller/Endpoint this was the way I came up with. The way it was implemented before if we try to use [Authorize] attribute with the Authentication settings disabled we will get an exception for trying to use Authorization without the service being registered.

Reply:
The question is how does the filter run if there are multiple authorization filters installed. Does the order matter?

Reply:
So Eric this is not a Filter it is a Handler and we can have many instances as we want to all of them are going to be executed. 

https://docs.microsoft.com/en-us/aspnet/core/security/authorization/policies?view=aspnetcore-3.1

```
public async Task<AuthorizationResult> AuthorizeAsync(ClaimsPrincipal user, 
             object resource, IEnumerable<IAuthorizationRequirement> requirements)
{
    // Create a tracking context from the authorization inputs.
    var authContext = _contextFactory.CreateContext(requirements, user, resource);

    // By default this returns an IEnumerable<IAuthorizationHandlers> from DI.
    var handlers = await _handlers.GetHandlersAsync(authContext);

    // Invoke all handlers.
    foreach (var handler in handlers)
    {
        await handler.HandleAsync(authContext);
    }

    // Check the context, by default success is when all requirements have been met.
    return _evaluator.Evaluate(authContext);
}
```